Card/src/main/java/com/braintreepayments/api/card/BinData.kt

```diff
@@ -0,0 +1,147 @@
+package com.braintreepayments.api.card
+
+import android.os.Parcelable
+import androidx.annotation.RestrictTo
+import androidx.annotation.StringDef
+import com.braintreepayments.api.sharedutils.Json
+import kotlinx.parcelize.Parcelize
+import org.json.JSONObject
+
+/**
+ * A class to contain BIN data for the card number
+ */
+@Parcelize
+data class BinData(
+    private var prepaid: String = UNKNOWN,
+    private var healthcare: String = UNKNOWN,
+    private var debit: String = UNKNOWN,
+    private var durbinRegulated: String = UNKNOWN,
+    private var commercial: String = UNKNOWN,
+    private var payroll: String = UNKNOWN,
+    private var issuingBank: String = UNKNOWN,
+    private var countryOfIssuance: String = UNKNOWN,
+    private var productId: String = UNKNOWN
+) : Parcelable {
+
+    @Retention(AnnotationRetention.SOURCE)
+    @StringDef(YES, NO, UNKNOWN)
+    internal annotation class BinType
```

Comment:
Could we convert `BinType` to an enum type?

Reply:
Definitely yeah. We should convert all `@StringDef` and `@IntDef`, etc. types to `enum`. These are holdovers going way back to `v3`. At the time enums were considered memory-heavy in Android. I believe they fixed this at some point and it's no longer a concern.

Reply:
Will change to enum